/**:
  ros__parameters:
    robot_frame_id: "base_link"
    lidar_frame_id: "ouster"
    map_frame_id: "map"
    pub_global_plan_topic: "~/global_plan"
    pub_goal_point_viz_topic: "~/goal_point_viz"
    pub_trajectory_viz_topic: "~/traj_viz"
    pub_grid_viz_topic: "~/vdb_viz"
    pub_frontier_viz_topic: "~/frontier_viz"
    pub_clustered_frontier_viz_topic: "~/clustered_frontier_viz"
    sub_lidar_topic: "sensor_pointcloud"
    sub_map_topic: "vdb_map_visualization"
    sub_vdb_topic: "vdb_grid"
    sub_robot_tf_topic: "/tf"
    sub_target_path_topic: "/perspective_goals"
    srv_exploration_toggle_topic: "/robot_1/behavior/global_plan_toggle"

    publish_visualizations: false # should trajectory visualizations be published
    num_paths_to_generate: 5 # how many paths to string together

    # Planner Parameters
    max_start_to_goal_dist_m: 10.0 # how max distance the goal can be from the current robot position
    checking_point_cnt: 100 # how many points to collision check between the start and goal points
    max_z_change_m: 2.0 # max height that the goal can deviate from the current height
    collision_padding_m: 0.5 # how much space should the path have to any obstacles
    path_end_threshold_m: 0.2 # how close to the goal will the path be considered complete
    max_yaw_change_degrees: 180.0 # how much the goal can deviate from the current orientation

    # Mapping Parameters
    map_voxel_resolution: 0.25 # voxel grid size
    l_occ: 1.0
    cluster_cube_dim: 5.0
    voxel_cluster_count_thresh: 50

    # Stall detection parameters
    position_change_threshold: 0.1  # Minimum distance (meters) to consider as movement
    stall_timeout_seconds: 20.0     # Time without movement before clearing plan

    # view sampling
    bbox_length                         : 1.5
    bbox_breadth                        : 1.5
    bbox_height                         : 1.5
    viewing_distance                    : 1.5 # not used
    num_viewpts_per_cluster             : 6
    kViewingDistanceInnerR              : 1.5
    kViewingDistanceOuterR              : 2.0
    kViewingDistanceZOffset             : 1.0
    viewp_bbox_unknown_frac_thresh      : 1.0 # In a box around a voxel, all unknown as free, all are allowed to be unknown
    viewp_bbox_known_occ_thresh         : 0.0 # No occupation allowed
    coll_check_endpoint_offset          : 0.4
    robot_model_bbox_fraction           : 0.1

    # Traj Planning
    too_close_distance: 4.0
    too_close_penalty: 5.0
    odometry_match_distance: 3.0
    odometry_match_angle: 3.14159
    odometry_match_penalty: 1000.0
    momentum_collision_check_distance: 15.0
    momentum_collision_check_distance_min: 4.0
    momentum_change_max_reward: 6.0
    momentum_change_collision_reward: 2.0
    viewpoint_neighborhood_too_close_radius: 10.0
    viewpoint_neighborhood_sphere_radius: 6.0
    viewpoint_neighborhood_count: 5
    momentum_minimum_distance: 0.5
    momentum_time: 6.0
    momentum_collision_check_step_size: 0.4

    # Voxxel Planner RRT params
    planner_norm_limit: 1.0
    dense_step: 0.1
    max_explore_dist: 100.0
    planner_max_iter: 20000
    traj_smooth_horizon: 50
    max_connection_iter: 1000

    # RRT force adding nodes around the start and goal
    rrt_region_nodes_count: 16
    rrt_region_nodes_radius: 0.4
    rrt_region_nodes_z_layers_count_up: 1
    rrt_region_nodes_z_layers_step: 0.4

    priority_level_thred: 10

    # Collision checker (Safe)
    robot_model_safer_cube_dim: 1.8

    # Collision checker (Risky)
    robot_model_cube_dim: 1.4 #0.8